server/service/osquery_utils/queries.go

```diff
@@ -1604,6 +1622,9 @@ func GetDetailQueries(
 
 	if appConfig != nil && appConfig.MDM.EnabledAndConfigured {
 		for key, query := range mdmQueries {
+			if slices.Equal(query.Platforms, []string{"windows"}) && !appConfig.MDM.WindowsEnabledAndConfigured {
```

Comment:
not super common, but what if a query can run on darwin and windows? what if we use `slices.Contains` instead?

```suggestion
			if slices.Contains(query.Platforms, "windows") && !appConfig.MDM.WindowsEnabledAndConfigured {
```

Reply:
Yeah, I considered that and figured that we'd cross that bridge if/when we get there. This logic only applies to the MDM detail queries and this is the only one of the MDM detail queries that pertains to Windows currently.

Reply:
sounds good! no request to change it on my side 👍 

for the sake of argument, my take would be: why don't do something that works for all cases now? the logic you're applying makes sense to me when the "general" solution adds complexity or is more elaborated than the "single case" solution.

Reply:
I definitely see your point and appreciate the discussion :) 

If we generalize the test, I suppose the ideal outcome would be that it is more likely to fail in cases where a new dual-platform query is added. That way the engineer will be forced to think about what rule should be applied to that case. 

What gave me pause about trying to generalize is that the thing we're needing to test here is that a Windows-only MDM query is included for Windows hosts only if Windows MDM is enabled. If a hypothetical MDM query really applies to both platforms, how sure can be be now that we'd want the same rule to apply? 

What do you think?